browser/ui/views/frame/split_view/brave_multi_contents_view.cc

```diff
@@ -54,11 +54,12 @@ void BraveMultiContentsView::UseContentsContainerViewForWebPanel() {
   }
 }
 
-void BraveMultiContentsView::SetWebPanelVisible(bool visible) {
+void BraveMultiContentsView::SetWebPanelContents(
+    content::WebContents* web_contents) {
   CHECK(contents_container_view_for_web_panel_);
-  contents_container_view_for_web_panel_->SetVisible(visible);
-  contents_container_view_for_web_panel_->UpdateBorderAndOverlay(true, true,
-                                                                 false);
+  contents_container_view_for_web_panel_->contents_view()->SetWebContents(
+      web_contents);
+  contents_container_view_for_web_panel_->SetVisible(!!web_contents);
```

Comment:
suggestion: no need to explicitly cast contents to boolean?

```suggestion
  contents_container_view_for_web_panel_->SetVisible(web_contents);
```